### PR TITLE
Add docs for new team access API behavior

### PR DESCRIPTION
**Pull request by jondavidjohn**
Documentation for the pending release of added API behavior

### CODE REVIEW

File: content/source/docs/cloud/api/team-access.html.md [hunk 1]
```diff
@@ -41,9 +43,13 @@ Status  | Response                                          | Reason
 
 [These are standard URL query parameters](./index.html#query-parameters); remember to percent-encode `[` as `%5B` and `]` as `%5D` if your tooling doesn't automatically encode URLs.
 
+This endpoint supports pagination [with standard URL query parameters](./index.html#query-parameters).  If neither pagination query parameters are provided, the endpoint will not be paginated and will return all results.
```
Comment (dylanegan) [inline content/source/docs/cloud/api/team-access.html.md:46]: 👨‍🍳😙 🖐

**Review by dylanegan — APPROVED** [review]

File: content/source/docs/cloud/api/team-access.html.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ The team access APIs are used to associate a team to permissions on a workspace.
 
 -> **Note**: A `team-workspace` resource represents a team's _local_ permissions on a specific workspace. Teams can also have _organization-level_ permissions that grant access to workspaces, and Terraform Cloud uses whichever access level is higher. (For example: a team with the "manage workspaces" permission has admin access on all workspaces, even if their `team-workspace` on a particular workspace only grants read access.) For more information, see [Managing Workspace Access](../users-teams-organizations/teams.html#managing-workspace-access).
 
+Any member of an organization can view team access relative to their own team memberships, this includes secret teams of which they are a member. Organization owners and workspace admins can modify team access or view the full set of secret team accesses. The organization token and the owners team token can act as an owner on these endpoints. ([More about permissions.](/docs/cloud/users-teams-organizations/permissions.html))
```
Comment (vaficionado) [inline content/source/docs/cloud/api/team-access.html.md:31]:
```suggestion
Any member of an organization can view team access relative to their own team memberships, including secret teams of which they are a member. Organization owners and workspace admins can modify team access or view the full set of secret team accesses. The organization token and the owners team token can act as an owner on these endpoints. ([More about permissions.](/docs/cloud/users-teams-organizations/permissions.html))
```

**Review by vaficionado — COMMENTED** [review]
One minor tweak, otherwise LGTM!

File: content/source/docs/cloud/api/team-access.html.md (hunk 2 above)
Comment (anamahalo) [inline content/source/docs/cloud/api/team-access.html.md:31]:
```suggestion
Any member of an organization can view team access relative to their own team memberships -- this includes secret teams of which they are a member. Organization owners and workspace admins can modify team access or view the full set of secret team accesses. The organization token and the owners team token can act as an owner on these endpoints. ([More about permissions.](/docs/cloud/users-teams-organizations/permissions.html))
```

Comment (anamahalo) [reply content/source/docs/cloud/api/team-access.html.md:31]: Alternative option of a `hyphen` if you had wanted this point to stand out a bit more but still remain on a continuum from prior part of the sentence.

**Review by anamahalo — APPROVED** [review]
+1 to the comma splice fix.  Provided a single alternative solution in the off-chance it aligns more with intention toward the reader.

**Pull request merged**